frontend/packages/kubevirt-plugin/integration-tests/tests/mocks/vmBuilderPresets.ts

```diff
@@ -0,0 +1,124 @@
+import { deepFreeze } from '../utils/utils';
+import { VMBuilder } from '../models/vmBuilder';
+import * as Combinatorics from 'js-combinatorics';
+import { Flavor, Workload, OperatingSystem, ProvisionSource } from '../utils/constants/wizard';
+import { VMBuilderDataGenerationConfig, VMBuilderData } from '../types/vm';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import {
+  provisionSources,
+  rootDisk,
+  multusNetworkInterface,
+  containerRootDisk,
+  flavorConfigs,
+  getDiskToCloneFrom,
+} from './mocks';
+import { VirtualMachine } from '../models/virtualMachine';
+import { VMTemplateBuilder } from '../models/vmtemplateBuilder';
+import { VirtualMachineTemplate } from '../models/virtualMachineTemplate';
+
+const generateBuilders = (baseBuilder, groupIndex, keys, values) => {
+  const newComputedBuilders: { [k: string]: VMBuilder } = {};
+  Combinatorics.cartesianProduct(...values)
+    .toArray()
+    .forEach((additionalBuilderValues, additionalBuilderId) => {
+      const additionalBuilderData = additionalBuilderValues.reduce((acc, value, keyIdx) => {
+        acc[keys[keyIdx]] = value;
+        return acc;
+      }, {} as VMBuilderData);
+      const additionalBuilder = new VMBuilder().setData(additionalBuilderData);
+      newComputedBuilders[`config ${groupIndex}-${additionalBuilderId}`] = new VMBuilder(
+        baseBuilder,
+      )
+        .setBuilder(additionalBuilder)
+        .generateName(`${additionalBuilderId}`);
+    });
+  return newComputedBuilders;
+};
+
+export const computedVMBuilders = deepFreeze(
```

Comment:
and some doc here? what is the result? You can just write some small example.

Reply:
are we using the generated builders somewhere?

Reply:
We're not using these generated builders atm. I'm not sure how to deal with adding polarion IDs for this. We could leave it without an ID and not have a test case in polarion, or create one test case in Polarion and describe the matrix of created VMs there. I'm afraid that having test case in Polarion for each combination is impossible to maintain givan the interface Polarion provides us....

We could use them in v2v suite, which is all about importing differently configured VMs

Reply:
I think having one test case with the matrix could serve this purpose quite fine. Can we use it? Or are there some issues with it?

+1 for v2v

Reply:
Don't see any issue, should I include it in this PR?

Reply:
IMO you can, but up to you.